core/vibe/core/drivers/libevent2.d

```diff
@@ -389,8 +389,29 @@ final class Libevent2Driver : EventDriver {
 
 	Libevent2TCPListener listenTCP(ushort port, void delegate(TCPConnection conn) @safe connection_callback, string address, TCPListenOptions options)
 	{
-		auto bind_addr = resolveHost(address, AF_UNSPEC, false);
-		bind_addr.port = port;
+		NetworkAddress bind_addr;
+		version(Posix)
+		{
+			import core.sys.posix.sys.un;
+			import core.stdc.string : strcpy;
+
+			if (address[0] == '/')
+			{
+				bind_addr.family = AF_UNIX;
+				sockaddr_un* s = bind_addr.sockAddrUnix();
+				enforce(s.sun_path.length > address.length, "Unix sockets cannot have that long a name.");
+				s.sun_family = AF_UNIX;
+				() @trusted { strcpy(cast(char*)s.sun_path.ptr,address.toStringz()); } ();
```

Comment:
The `toStringz` seems to be unnecessary here. Why don't you simply copy it over (e.g. `s[] = address[]`) and then set the next char to `\0`?

Reply:
Also copied from the UDS implementation of requestHTTP.

Reply:
`s.sun_path` is a `byte[]`. It could be changed to `() @trusted { s.sun_path[0 .. address.length] = (cast(byte[])address)[]; s.sun_path[address.length] = '\0'; } ();`, which is more or less [what Phobos does](https://github.com/dlang/phobos/blob/master/std/socket.d#L1988). Would that be better?

Reply:
Would be better in many cases because it always avoids a GC allocation, while `toStringz` needs to allocate if the string is not already zero terminated in memory for some reason (e.g. string literal or in a buffer that is padded with zeros).